test/apis/v1alpha1/api_validation_integration_test.go

```diff
@@ -0,0 +1,899 @@
+			Expect(statusErr.Status().Message).Should(ContainSubstring("a lowercase RFC 1123 subdomain"))
+		})
+
+		It("should deny creating API with invalid name ending with non-alphanumeric character", func() {
+			var nameEndingWithNonAlphanum = "abcdef-123456789-123456789-123456789-123456789-123456789-"
+
+			// Create the API.
+			multiClusterServiceName := &v1alpha1.MultiClusterService{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      nameEndingWithNonAlphanum,
+					Namespace: testNamespace,
+				},
+				Spec: v1alpha1.MultiClusterServiceSpec{
+					ServiceImport: v1alpha1.ServiceImportRef{
+						Name: "service-import-name",
+					},
+				},
+			}
+			By(fmt.Sprintf("expecting denial of CREATE API %s", nameEndingWithNonAlphanum))
+			err := hubClient.Create(ctx, multiClusterServiceName)
+			var statusErr *k8serrors.StatusError
+			Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Create API call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8serrors.StatusError{})))
+			Expect(statusErr.Status().Message).Should(ContainSubstring("a lowercase RFC 1123 subdomain"))
+		})
+
+		It("should deny creating API with invalid name containing character that is not alphanumeric and not -", func() {
+			var nameContainsUnderscore = "a_bcdef-123456789-123456789-123456789-123456789-123456789-123456789-123456789"
+
+			// Create the API.
+			multiClusterServiceName := &v1alpha1.MultiClusterService{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      nameContainsUnderscore,
+					Namespace: testNamespace,
+				},
+				Spec: v1alpha1.MultiClusterServiceSpec{
+					ServiceImport: v1alpha1.ServiceImportRef{
+						Name: "service-import-name",
+					},
+				},
+			}
+			By(fmt.Sprintf("expecting denial of CREATE API %s", nameContainsUnderscore))
+			err := hubClient.Create(ctx, multiClusterServiceName)
+			var statusErr *k8serrors.StatusError
+			Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Create API call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8serrors.StatusError{})))
+			Expect(statusErr.Status().Message).Should(ContainSubstring("a lowercase RFC 1123 subdomain"))
+		})
+	})
+
+	Context("Test MultiClusterService creation API validation - valid cases", func() {
+		It("should allow creating API with valid name size", func() {
+			var nameValid = "abc-123456789-123456789-123456789-123456789-123456789-123456789"
+
+			// Create the API.
+			multiClusterServiceName := &v1alpha1.MultiClusterService{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      nameValid,
+					Namespace: testNamespace,
+				},
+				Spec: v1alpha1.MultiClusterServiceSpec{
+					ServiceImport: v1alpha1.ServiceImportRef{
+						Name: "service-import-name",
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, multiClusterServiceName)).Should(Succeed())
```

Comment:
```suggestion
			Expect(hubClient.Create(ctx, multiClusterServiceName)).Should(Succeed(), "failed to create multiClusterService")
```
missed this before, feel better to add the debug information here